include/glow/Backends/DeviceManager.h

```diff
@@ -152,6 +152,27 @@ class DeviceManager {
   /// \returns the DeviceInfo for this device containing peak limits for
   /// compute and bandwidths (used in partitioning).
   virtual DeviceInfo getDeviceInfo() const { return DeviceInfo(); }
+
+  /// Copies the contents of \p tensor from the host to the \p location address
+  /// on this device. Updates the tensor residency info.
+  virtual bool transferToDevice(Tensor &tensor, void *context) {
```

Comment:
This updates the tensor residency info to the information in void *context right? 
Should context be named residencyInfo or location?
Where does the location information come from? Should there be a getAddress() method as well?

Reply:
Personally, I think it should be called location and should definitely be optional.

Don't think getAddress() works because some devices may not use addresses.

Reply:
Device Residency Info contains pointers to:
1) DeviceManager
2) Opaque context
So it is not the residencyInfo itself.
In addition, I think there may be use cases where the context is useful to store additional information besides the actual memory address / location. 
For example, let's consider a backend with a memory pool for tensors. Its device manager should have a mapping between keys and buffers. If a tensor becomes device resident, it should get a buffer. In this case, one option is to store that key in the context.

Naming - how about locationContext?

GetAddress(...) : this goes to how a certain device manager creates a new locationContext.
The challenge here is to figure out if there is a single interface that fits most backends, i.e. if it makes sense to add createLocationContext() (taking no input arguments). Or does a certain backend needs the function name? or the placeholder name? (as we briefly started discussing :))

Reply:
I'd be concerned that if we find an approach that works well for all backends we know about now, we'll encounter a future backend that doesn't fit that scheme.